sqlmesh/core/model/kind.py

```diff
@@ -407,7 +362,42 @@ class IncrementalByTimeRangeKind(_IncrementalBy):
     time_column: TimeColumn
     auto_restatement_intervals: t.Optional[SQLGlotPositiveInt] = None
 
-    _time_column_validator = TimeColumn.validator()
+    @field_validator("time_column", mode="before")
+    @classmethod
+    def _time_column_validator(cls, v: t.Any, values: t.Any) -> TimeColumn:
+        dialect = get_dialect(values)
```

Comment:
Shouldn't this be `values.data`? The fact that our tests didn't catch this is kind of scary

Reply:
It seems like the `get_dialect` implementation accounts for both `ValidationInfo` and `dict`, see [here](https://github.com/TobikoData/sqlmesh/blob/568b265355af6d35f0055831cdec0f70a75fff82/sqlmesh/utils/pydantic.py#L45-L57).

Reply:
Changed it anyway though.